src/Microsoft.AspNetCore.OData/Batch/ODataBatchReaderExtensions.cs

```diff
@@ -140,9 +146,19 @@ public static class ODataBatchReaderExtensions
 
             foreach (var header in batchRequest.Headers)
             {
-                // Copy headers from batch, overwriting any existing headers.
                 string headerName = header.Key;
                 string headerValue = header.Value;
+
+                if (headerName.Trim().ToLowerInvariant() == "prefer")
+                {
+                    // in the case of Prefer header, we don't want to overwrite,
+                    // instead we merge preferences defined in the individual request with those inherited from the batch
+                    request.Headers.TryGetValue(headerName, out StringValues batchReferences);
+                    request.Headers[headerName] = MergeIndividualAndBatchPreferences(headerValue, batchReferences);
+                    continue;
+                }
+
+                // Copy headers from batch, overwriting any existing headers.
                 request.Headers[headerName] = headerValue;
```

Comment:
I think it's a little easier to read if you remove the continue on line 158 and put line 162 in an else.